testapi.pm

```diff
@@ -2127,13 +2127,16 @@ sub diag (@args) { bmwqemu::diag(@args) }
 Return VM's host IP.
 In a kvm instance you reach the VM's host under default 10.0.2.2
 
+Optional C<$force_worker_hostname> argument will force using actual worker IP
+even on KVM guests.
+
 =cut
 
-sub host_ip () { check_var('BACKEND', 'qemu') ? get_var('QEMU_HOST_IP', '10.0.2.2') : get_required_var('WORKER_HOSTNAME') }
+sub host_ip ($force_worker_hostname = 0) { (check_var('BACKEND', 'qemu') && !$force_worker_hostname) ? get_var('QEMU_HOST_IP', '10.0.2.2') : get_required_var('WORKER_HOSTNAME') }
```

Comment:
It'll likely be called like `autoinst_url($path, $query, 1)` on your side. Unfortunately the meaning of `1` is not very clear then. So maybe make it a named parameter?

```suggestion
sub host_ip ($args = {}) { (check_var('BACKEND', 'qemu') && !$args->{force_worker_hostname}) ? get_var('QEMU_HOST_IP', '10.0.2.2') : get_required_var('WORKER_HOSTNAME') }
```